src/core/crypto/utils.ts

```diff
@@ -0,0 +1,24 @@
+import { HexInput } from "../../types";
+import { Hex } from "../hex";
+
+/**
+ * Helper function to convert a message to sign or to verify to a valid message input
+ *
+ * @param message a message as a string or Uint8Array
+ *
+ * @returns a valid HexInput - Hex string or Uint8Array
+ */
+export const convertMessage = (message: HexInput): HexInput => {
+  // if message is of type string, verify it is a valid Hex string
+  if (typeof message === "string") {
+    const isValid = Hex.isValid(message);
+    // If message is not a valid Hex string, convert it into a Buffer
+    if (!isValid.valid) {
+      return Buffer.from(message, "utf8");
+    }
+    // If message is a valid Hex string, return it
+    return message;
+  }
+  // message is a Uint8Array
+  return message;
+};
```

Comment:
Trying to come up with a better name than `convertMessage`

So, this lets you take in:

1. Hex string
2. A UTF-8 encoded string (as long as it isn't a hex string)
3. a Uint8array

Let's put that comment there, and maybe change it to... convertSigningMessage

Reply:
yea I wasnt sure what name makes sense. convertSigningMessage sounds good

From TS point of view, it lets you pass in `string | Uint8array` where `string` can be any string (regular string, hex string, etc)